ElrondNetwork/elrond-go
# EN-13187: add epoch start data endpoint

## Reasoning behind the pull request 
- an endpoint for returning the epoch start data for a given epoch was necessary
  
## Proposed Changes
- added a Rest API endpoint that will return some fields about the epoch start block for a provided epoch

## Testing procedure
When running a testnet (without DB removal) with this branch, on a given epoch (let's say epoch 7) try the following endpoints on a shard node and a meta node:
- `<node-ip>:<port>/node/epoch-start/0` ,  `<node-ip>:<port>/node/epoch-start/1` ... `<node-ip>:<port>/node/epoch-start/5` - for validators, it should work for current epoch minus 4 epochs (so if checking in epoch 7, it shouldn't work for epochs 1 and 2 - epoch 0 should still work). for observers, all the epochs should work
